### PR TITLE
Remove archived locations from case sharing groups

### PR DESCRIPTION
**Pull request by emord**
http://manage.dimagi.com/default.asp?189346#1063290

DB tests are so slow. :sob:

@esoergel 

@calellowitz I think you were working on location archiving. Not sure if this conflicts with your stuff

### CODE REVIEW

File: corehq/apps/locations/tests/test_location_types.py [hunk 1]
```diff
@@ -114,6 +114,20 @@ def test_only_case_sharing_descendents_included(self):
...
         )
 
+    def test_archived_locations_are_not_included(self):
+        parent_type = make_loc_type('parent', domain=self.domain, shares_cases=True, view_descendants=True)
+        child_type = make_loc_type('child', domain=self.domain, shares_cases=False)
+        grandchild_type = make_loc_type('grandchild', domain=self.domain, shares_cases=True)
+        parent_loc = make_loc('parent', type=parent_type.name, domain=self.domain)
+        child_loc = make_loc('child', type=child_type.name, domain=self.domain, parent=parent_loc)
+        grandchild_loc = make_loc('grandchild', type=grandchild_type.name, domain=self.domain, parent=child_loc)
+        archived_grandchild_loc = make_loc('archived_grandchild', type=grandchild_type.name, domain=self.domain, parent=child_loc, is_archived=True)
```
Comment (dimagimon) [inline corehq/apps/locations/tests/test_location_types.py]: F841 local variable 'archived_grandchild_loc' is assigned to but never used

Comment (dimagimon) [inline corehq/apps/locations/tests/test_location_types.py]: E501 line too long (148 > 115 characters)

File: corehq/apps/locations/tests/util.py [hunk 2]
```diff
@@ -9,9 +9,9 @@
 TEST_LOCATION_TYPE = 'location'
 
 
-def make_loc(code, name=None, domain=TEST_DOMAIN, type=TEST_LOCATION_TYPE, parent=None):
+def make_loc(code, name=None, domain=TEST_DOMAIN, type=TEST_LOCATION_TYPE, parent=None, is_archived=False):
     name = name or code
-    loc = Location(site_code=code, name=name, domain=domain, location_type=type, parent=parent)
+    loc = Location(site_code=code, name=name, domain=domain, location_type=type, parent=parent, is_archived=is_archived)
```
Comment (dimagimon) [inline corehq/apps/locations/tests/util.py]: E501 line too long (120 > 115 characters)

**Comment by czue** [comment]
> DB tests are so slow. :sob:

I see you're not using the optimized test runner (or you would have added dependencies). you should really use it. they can be super fast if you're only syncing a couple apps

**Pull request merged**